packages/types/src/types/requests/web-server/ProjectRequest.ts

```diff
@@ -3,14 +3,15 @@
  * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
  */
 
-import { Project } from '../../models';
+import { Entity, Project } from '../../models';
 import { KeysToCamelCase } from '../../../utils/casing';
 
 export interface Params {
   projectCode: string;
 }
 export type ProjectResponse = KeysToCamelCase<Project> & {
   name: string;
+  names?: Entity['name'][];
```

Comment:
I think `names` is already declared in `Project` type

Reply:
Not in `Project`, but it does show up in `SingleProject` from [`packages/tupaia-web/src/types/types.d.ts`](https://github.com/beyondessential/tupaia/blob/f55f09db440a1c30ae89ee9ac1cfa4fed11370fa/packages/tupaia-web/src/types/types.d.ts) (I can’t see a non-`.d` version of this file??)

Having now found that, I think you’re right that declaring it in `ProjectResponse` isn’t the way to go, but:

- when I log the `project` object from `useCurrentUser()`, it doesn’t have all the compulsory fields of `SingleProject` (so can’t cast)
- I’m not sure how to get the data flow right such that the `project` object that comes with `useCurrentUser()`

Might ask your help on this one tomorrow too 😅